lib/iris/coords.py

```diff
@@ -684,6 +766,13 @@ def _as_defn(self):
                          self.units, self.attributes, self.coord_system)
         return defn
 
+    # Must supply __hash__ as Python 3 does not enable it if __eq__ is defined.
+    # NOTE: Violates "objects which compare equal must have the same hash".
+    # Currently needed, but not really correct and should be fixed.
```

Comment:
But in fact I tested what happens when you error its usage, and the resulting errors are ***not*** about merge after all.
See : https://github.com/SciTools/iris/pull/1772#issuecomment-302080133

? I could say e.g. "needed so we can use Coords as set members and dictionary keys" ?